lib/src/git.rs

```diff
@@ -548,21 +546,18 @@ pub fn fetch(
     // git import --reset BRANCH`.
     // TODO: Once the command described above exists, it should be mentioned in `jj
     // help branch forget`.
-    let nonempty_branches = mut_repo
+    let nonempty_branches: HashSet<_> = mut_repo
         .view()
         .branches()
         .iter()
         .filter_map(|(branch, target)| target.local_target.as_ref().map(|_| branch.to_owned()))
-        .collect_vec();
+        .collect();
```

Comment:
the nittiest of nits but I like the turbofish better :upside_down_face: 

(`.collect::<HashSet<_>>()`)

Reply:
Hmm, I prefer type annotation on variable because it's easier to spot.

> in fact even the collect_vec while nice, is not something I can't live without

Agreed.